components/support/sync15-traits/src/server_timestamp.rs

```diff
@@ -118,8 +130,16 @@ mod test {
         let ser = serde_json::to_string(&ts).unwrap();
         assert_eq!("123.456".to_string(), ser);
 
-        // test deserialize
+        // test deserialize of float
         let ts: ServerTimestamp = serde_json::from_str(&ser).unwrap();
         assert_eq!(ServerTimestamp(123_456), ts);
+
+        // test deserialize of whole number
+        let ts: ServerTimestamp = serde_json::from_str("123").unwrap();
+        assert_eq!(ServerTimestamp(123_000), ts);
+
+        // test deserialize of negative number
+        let ts: ServerTimestamp = serde_json::from_str("-123").unwrap();
```

Comment:
Did these tests fail before the change? (If so: that's really surprising, what a footgun!)